test/conformance/enqueue/urEnqueueUSMMemset2D.cpp

```diff
@@ -1,2 +1,215 @@
+        width = std::get<1>(GetParam()).width;
+        height = std::get<1>(GetParam()).height;
+        num_elements = pitch * height;
+        host_mem = std::vector<char>(num_elements);
+
+        bool device_usm{false};
+        ASSERT_SUCCESS(urDeviceGetInfo(device, UR_DEVICE_INFO_USM_DEVICE_SUPPORT,
+                                       sizeof(bool), &device_usm, nullptr));
+
+        if (!device_usm) {
+            GTEST_SKIP_("Device USM is not supported");
+        }
+
+        ur_usm_mem_flags_t flags{};
+        ASSERT_SUCCESS(urUSMDeviceAlloc(context, device, &flags, num_elements, 0,
+                                        reinterpret_cast<void **>(&ptr)));
+        ASSERT_SUCCESS(urQueueFlush(queue));
+    }
+
+    void TearDown() override {
+        if (ptr) {
+            EXPECT_SUCCESS(urUSMFree(context, ptr));
+        }
+
+        UUR_RETURN_ON_FATAL_FAILURE(urQueueTestWithParam::TearDown());
+    }
+
+    bool verifyData() {
+        EXPECT_SUCCESS(
+            urEnqueueUSMMemcpy2D(queue, true, host_mem.data(), pitch, ptr, pitch,
+                                 width, height, 0, nullptr, nullptr));
+        for (int w = 0; w < width; ++w) {
+            for (int h = 0; h < height; ++h) {
+                char *host_ptr = host_mem.data();
+                size_t index = (pitch * h) + w;
+                if (*(host_ptr + index) != memset_value) {
+                    return false;
+                }
+            }
+        }
+        return true;
+    }
+
+    const int memset_value = 12;
+    int pitch;
+    int width;
+    int height;
+    uint32_t num_elements;
+    std::vector<char> host_mem;
+    int *ptr{nullptr};
+};
+
+std::vector<uur::TestParameters2D> test_cases{
+    /* Everything set to 1 */
+    {1, 1, 1},
+    /* Height == 1 && Pitch > width */
+    {1024, 256, 1},
+    /* Height == 1 && Pitch == width */
+    {1024, 1024, 1},
+    /* Height > 1 && Pitch > width */
+    {1024, 256, 256},
+    /* Height > 1 && Pitch == width + 1 */
+    {234, 233, 23},
+    /* Height == 1 && Pitch == width + 1 */
+    {234, 233, 1}};
+
+UUR_TEST_SUITE_P(urEnqueueUSMMemset2DTestWithParam,
+                 testing::ValuesIn(test_cases),
+                 uur::print2DTestString<urEnqueueUSMMemset2DTestWithParam>);
+
+TEST_P(urEnqueueUSMMemset2DTestWithParam, Success) {
+
+    ur_event_handle_t event = nullptr;
+    ASSERT_SUCCESS(
+        urEnqueueUSMMemset2D(queue, ptr, pitch, memset_value, width, height, 0,
+                             nullptr, &event));
+    EXPECT_SUCCESS(urQueueFlush(queue));
+
+    ASSERT_SUCCESS(urEventWait(1, &event));
+    ur_event_status_t event_status;
+    EXPECT_SUCCESS(urEventGetInfo(event, UR_EVENT_INFO_COMMAND_EXECUTION_STATUS,
+                                  sizeof(ur_event_status_t), &event_status,
+                                  nullptr));
+    ASSERT_EQ(UR_EVENT_STATUS_COMPLETE, event_status);
+    EXPECT_SUCCESS(urEventRelease(event));
+
+    verifyData();
```

Comment:
```suggestion
    ASSERT_TRUE(verifyData());
```

Reply:
I updated this. But I'm using ASSERT_NO_FATAL_FAILURE instead of ASSERT_TRUE. The reason is that I'm asserting inside the function as well which would do nothing. 

I think this is a better way than, for example, returning a bool. If there is a failure on one of the assertions, it will be easy to pinpoint if it's the memcpy or the checks that are failing.